Factor

Lcjacobpd opened the pull request:
Integration of factor evaluator code

Lcjacobpd committed:
build & run readme

+ added README.md to build-and-run example

Commit:
Squashed commit of the following:

commit 26ca229bd0e50a2cb6d23eecd9d6a2eda984f9e1
Author: Jacob Dickens <lcjacobpd@localhost.localdomain>
Date:   Fri Oct 2 09:31:35 2020 -0600

    Resized Logos

    = resized bueno logos in docs/img/

commit ba43937ffc23e516aea8c3d79a802f0dcecc3a3f
Author: Jacob Dickens <lcjacobpd@localhost.localdomain>
Date:   Fri Oct 2 09:13:51 2020 -0600

    Bueno Logo

    + added 3 logos to docs/img/
    + added logo think to main readme

Commit:
Squashed commit of the following:

commit 26ca229bd0e50a2cb6d23eecd9d6a2eda984f9e1
Author: Jacob Dickens <lcjacobpd@localhost.localdomain>
Date:   Fri Oct 2 09:31:35 2020 -0600

    Resized Logos

    = resized bueno logos in docs/img/

commit ba43937ffc23e516aea8c3d79a802f0dcecc3a3f
Author: Jacob Dickens <lcjacobpd@localhost.localdomain>
Date:   Fri Oct 2 09:13:51 2020 -0600

    Bueno Logo

    + added 3 logos to docs/img/
    + added logo think to main readme

Commit:
Merge with Upstream

Commit:
Merge branch 'master' of github.com:Lcjacobpd/bueno

Commit:
Resolve Merge and SVG

+ resolved merge conflict in examples/build-and-run/README.md
+ added svg file for icon in docs/img

Commit:
build-and-run/README.md

+ pulled markdown from upstream (discard local)

Commit:
Factor

+ added testing/factor directory
  : README.md
  : factor.py
  : experiment directory

+ integreated Factor class in experiment.py

samuelkgutierrez commented:
Can you please resolve all merge conflicts?

Commit:
Merge remote-tracking branch 'upstream/master'

samuelkgutierrez commented:
@Lcjacobpd can you now make sure that all checks pass?

Commit:
Minor style fix

+ fixed uncompliant line

Commit:
Typing fix

+ specified return type properly
+ passed string to logger.log instead of list

Commit:
bueno/tests

+ mirrored style compliance changes in bueno/tests/factor
  : updated factor.py
  : updated experiements/test.py

Commit:
logger.log

+ removed instances of emlog with end=''
  : replaced with print(###, end='')

samuelkgutierrez commented on bueno/public/experiment.py:
```diff
@@ -522,4 +522,164 @@ def runcmds(
+            # Case 1: List is 1 item too long
+            # Combine the first 2 items
+            if length == (self.dimensions + 1):
+                print('Case 1 -> ', end='')
```
Please remove all debug print statements.

samuelkgutierrez commented on tests/factor/factor.py:
```diff
@@ -0,0 +1,210 @@
+    return args
+
+
+class Factor:
```
Why is there a lot of seemingly duplicate code here? Can you reuse the code that is in  `public/experiment.py`?

Commit:
Request Fix

- removed duplicate code in tests/factor/factor.py
  : call experiment.py methods

- removed debug print statements from experiment.py

samuelkgutierrez reviewed (approved):
Thank you, Jacob. Looks good.
